course_info/views.py

```diff
@@ -37,6 +40,26 @@ def partials(request, path):
     return render(request, 'course_info/partials/' + path, {})
 
 
+def _get_canvas_roles():
+    """
+    Get the list of canvas roles from the canvas root account ('self') and
+    create a list in the format the front end wants.
+    :return:
+    """
+    roles = []
+    try:
+        canvas_roles = canvas_api_helpers_roles.get_roles_for_account_id('self')
+        user_roles = UserRole.objects.filter(pk__in=settings.ADD_PEOPLE_TO_COURSE_ALLOWED_ROLES_LIST)
+        for role in user_roles:
+            label = canvas_roles[role.canvas_role_id]['label']
+            roles.append({'roleId': role.role_id,'roleName': label})
+    except CanvasAPIError:
```

Comment:
Maybe we just want to `except:`, which would catch `UserRole` lookup and dict key lookup errors as well. Otherwise this `CanvasAPIError` except block should just be right under the `get_roles_for_account_id()` line. The broader `except` might be handy, though. The `logger.exception()` message in that case should also be broader.